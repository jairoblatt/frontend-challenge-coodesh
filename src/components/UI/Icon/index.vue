<template>
  <i v-if="iconSanitize" :class="`mdi ${iconSanitize}`" :style="styles"></i>
</template>

<script lang="ts">
import Vue, { PropOptions, PropType } from "vue";
import CSS from "csstype";

export default Vue.extend({
  name: "Icon",

  props: {
    size: {
      type: [String, Number],
      default: 14,
    } as PropOptions<string | number>,

    color: String as PropType<string>,

    icon: {
      type: String,
      default: "",
    } as PropOptions<string>,
  },

  computed: {
    iconSanitize(): string {
      return this.$slots.default?.[0]?.text || this.icon;
    },

    styles(): CSS.Properties {
      return {
        fontSize: this.size + "px",
        ...(this.color && { color: this.color }),
      };
    },
  },
});
</script>
