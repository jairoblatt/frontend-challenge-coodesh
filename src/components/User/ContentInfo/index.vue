<template>
  <div class="content__info">
    <span> <Icon :icon="icon" size="19" /> </span>
    <small>
      <slot />
    </small>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from "vue";

export default Vue.extend({
  props: {
    icon: String as PropType<string>,
  },
});
</script>

<style lang="postcss" scoped>
.content__info {
  @apply flex items-center justify-start md:justify-center border-solid border rounded-md border-gray-100 my-1 md:my-3 mx-2 md:w-auto w-full;
}

.content__info span {
  @apply font-bold py-1 px-3 bg-primary-base text-white;
}

.content__info small {
  @apply font-semibold px-4 text-base;
}
</style>
