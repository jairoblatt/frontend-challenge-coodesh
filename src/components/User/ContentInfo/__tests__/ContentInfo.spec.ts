import { mount, Wrapper, MountOptions, createLocalVue } from "@vue/test-utils";
import ContentInfo from "../index.vue";
import Icon from "@/components/UI/Icon/index.vue";

const localVue = createLocalVue();
localVue.component("Icon", Icon);

type Instance = InstanceType<typeof ContentInfo>;
let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;

describe("ContentInfo.vue ", () => {
  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(ContentInfo, {
        ...options,
        localVue,
      });
    };
  });

  it("Should mount the component", () => {
    const wrapper = mountFunction();

    expect(wrapper.vm).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should render default slot when him is passed", () => {
    const slot = `<h5>Something</h5>`;
    const wrapper = mountFunction({
      slots: {
        default: slot,
      },
    });

    expect(wrapper.find("h5").exists()).toBe(true);
  });

  it("Should render the current icon", () => {
    const wrapper = mountFunction({
      propsData: {
        icon: "mdi-user",
      },
    });

    expect(wrapper.findComponent(Icon).classes()).toEqual(["mdi", "mdi-user"]);
  });
});
