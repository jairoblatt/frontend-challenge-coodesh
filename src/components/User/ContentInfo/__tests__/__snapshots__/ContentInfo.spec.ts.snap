// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentInfo.vue  Should mount the component 1`] = `"<div class=\\"content__info\\"><span><!----></span> <small></small></div>"`;
